src/modules/flow/netctl/netctl.json

```diff
@@ -0,0 +1,80 @@
+{
+  "$schema": "http://solettaproject.github.io/soletta/schemas/node-type-genspec.schema",
+  "name": "netctl",
+  "meta": {
+    "author": "Intel Corporation",
+    "license": "Apache-2.0",
+    "version": "1"
+  },
+  "types": [
+    {
+      "category": "netctl",
+      "description": "Send list of all network services attached to board.",
+      "methods": {
+        "open": "open_list_services",
+	"close": "close_list_services"
+      },
+      "name": "netctl/list-services",
+      "out_ports": [
+        {
+          "data_type": "string",
+          "description": "Name of network services supported by system.",
+          "name": "OUT"
+        }
+      ],
+      "private_data_type": "list_service_data"
+    },
+    {
+      "category": "netctl",
+      "description": "Connect or disconnect to one available network access point.",
+      "in_ports": [
+        {
+          "data_type": "string",
+          "description": "Name of network AP.",
+          "methods": {
+            "process": "set_connect_ap_name"
+          },
+          "name": "CONNECT"
+        },
+        {
+          "data_type": "string",
+          "description": "Name of network AP.",
+          "methods": {
+            "process": "set_disconnect_ap_name"
+          },
+          "name": "DISCONNECT"
+        }
+      ],
+      "methods": {
+        "open": "open_service_connection",
+	"close": "close_service_connection"
+      },
+      "name": "netctl/connection",
+      "options": {
+        "members": [
+          {
+            "data_type": "string",
+            "default": null,
+            "description": "Name of AP to be used to connect. If 'null', no filter is applied.",
+            "name": "connect_ap"
```

Comment:
"AP" is not right. It should be "service". If the service is an access point (in wifi terms) or something else, it's not visible to the user here.